mewarpx/mewarpx/emission.py

```diff
@@ -107,10 +109,9 @@ def init_injectedparticles(self, fieldlist):
                 important; it must match the order for future particle appends
                 that are made.
         """
-        raise NotImplementedError
-        # self._injectedparticles_fields = fieldlist
-        # self._injectedparticles_data = warp.AppendableArray(
-        #     typecode='d', unitshape=[len(fieldlist)])
+        self._injectedparticles_fields = fieldlist
+        self._injectedparticles_data = appendablearray.AppendableArray(
+            typecode='d', unitshape=[len(fieldlist)])
 
     def record_injectedparticles(self, species, E_total=None, w=None,
                                  n=None):
```

Comment:
I think this whole function would be much cleaner as (untested though so there might be bugs).
```
    def record_injectedparticles(self, species, w, E_total=None):
        """Handles transforming raw particle information to the information
        used to record particles as a function of time. Also handles parallel
        sum and appending to the data array the current amount of injection.

        Note:
            Assumes the fixed form of fields given in Injector().  Doesn't
            check since this is called many times.
            Since a parallelsum is performed, call this with only the species
            argument if no particles are being added by this processor.

        Arguments:
            species: [TODO - adapt to WarpX]
            w (np.ndarray): Array of length npart with particle weights.
            E_total (np.ndarray): Array of length npart with E_total values.
            n (int): Number of macroparticles, _only_ needed if overriding the
                length of arrays This is useful mostly in the case that
                w and E_total are already summed over particles, in which case a
                single number can be passed for it rather than an array.
        """
        data = np.zeros(7)

        data[0] = mwxrun.get_it() * mwxrun.get_dt()
        data[1] = mwxrun.get_it()
        data[2] = species.species_number
        data[3] = self.getvoltage_e()
        data[4] = n if np.zise(w) == 1 else np.zise(w)
        data[5] = species.sq * np.sum(w)

        if E_total is not None:
              data[6] = np.sum(E_total)

        self.append_injectedparticles(data)
```